### PR TITLE
feat: resolve jsx files

### PR DESCRIPTION
**Pull request by Shou**
We resolve `.jsx` files such that they get handled by babel and bundled
with webpack.

Change-Type: patch
Changelog-Entry: Resolve JSX files

### CODE REVIEW

**Comment by Shou** [comment]
You can test this with a JSX file with the `jsx-components-svg-icon` branch

**Comment by jhermsmeier** [comment]
Can you elaborate a bit more on why we need this, what this changes, and how exactly this will work?

**Comment by Shou** [comment]
It's to tell Webpack to handle `.jsx` files and to treat them as modules, AFAICT it doesn't imply something else. If you try to import a JSX module with the config in `master` you just get an error right now saying it can't find the module.

File: webpack.config.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ module.exports = {
   module: {
     rules: [
       {
-        test: /\.js?$/,
+        test: /\.jsx?$/,
```
Comment (lurch) [inline webpack.config.js]: LOL, looks like the previous regex would have matched both `foo.j` and `foo.js` - was that intentional?
Whereas now it'll only match `foo.js` or `foo.jsx`

**Pull request merged**